src/Evolution/Executables/ScalarAdvection/EvolveScalarAdvection.hpp

```diff
@@ -0,0 +1,317 @@
+#include "Parallel/Actions/TerminatePhase.hpp"
+#include "Parallel/InitializationFunctions.hpp"
+#include "Parallel/PhaseControl/ExecutePhaseChange.hpp"
+#include "Parallel/PhaseControl/PhaseControlTags.hpp"
+#include "Parallel/PhaseControl/VisitAndReturn.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "ParallelAlgorithms/Actions/MutateApply.hpp"
+#include "ParallelAlgorithms/Events/Factory.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Actions/RunEventsAndTriggers.hpp"  // IWYU pragma: keep
+#include "ParallelAlgorithms/EventsAndTriggers/Completion.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/EventsAndTriggers.hpp"  // IWYU pragma: keep
+#include "ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Tags.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Trigger.hpp"
+#include "ParallelAlgorithms/Initialization/Actions/AddComputeTags.hpp"
+#include "ParallelAlgorithms/Initialization/Actions/RemoveOptionsAndTerminatePhase.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Sinusoid.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Time/Actions/AdvanceTime.hpp"            // IWYU pragma: keep
+#include "Time/Actions/ChangeSlabSize.hpp"         // IWYU pragma: keep
+#include "Time/Actions/ChangeStepSize.hpp"         // IWYU pragma: keep
+#include "Time/Actions/RecordTimeStepperData.hpp"  // IWYU pragma: keep
+#include "Time/Actions/SelfStartActions.hpp"       // IWYU pragma: keep
+#include "Time/Actions/UpdateU.hpp"                // IWYU pragma: keep
+#include "Time/StepChoosers/Factory.hpp"
+#include "Time/StepChoosers/StepChooser.hpp"
+#include "Time/StepControllers/Factory.hpp"
+#include "Time/StepControllers/StepController.hpp"
+#include "Time/Tags.hpp"
+#include "Time/TimeSequence.hpp"
+#include "Time/TimeSteppers/TimeStepper.hpp"
+#include "Time/Triggers/TimeTriggers.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/ErrorHandling/FloatingPointExceptions.hpp"
+#include "Utilities/MemoryHelpers.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <pup.h>
+
+/// \cond
+namespace Frame {
+struct Inertial;
+}  // namespace Frame
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace Parallel {
+template <typename Metavariables>
+class CProxy_GlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+template <size_t Dim, typename InitialData>
+struct EvolutionMetavars {
+  static constexpr size_t volume_dim = Dim;
+  using system = ScalarAdvection::System<Dim>;
+  using temporal_id = Tags::TimeStepId;
+  static constexpr bool local_time_stepping = false;
+
+  using initial_data = InitialData;
+  static_assert(
+      evolution::is_analytic_data_v<initial_data> xor
+          evolution::is_analytic_solution_v<initial_data>,
```

Comment:
We plan to move the choice of analytic solution or data to the input file, so you don't have to compile a different executable for every possible choice. Perhaps the scalar advection system would be a good opportunity to start this transition, since the system is new and doesn't have an overwhelming amount of code to re-factor. What do you think @kidder? Would you be willing to take this on @yoonso0-0?

Reply:
@nilsleiffischer Yes, I'd be happy to. My little concern is that I might be not familiar enough yet with the whole assembly of spectre executable -- if this plan is not something needs to be done very soon, I am willing to take.

Reply:
I think this transition can be done one executable at a time, so you can convert only the scalar advection system at first. The idea would be to have only one executable per dimension, `EvolveScalarAdvection{1,2,3}D`, and select the analytic solution, the analytic data, or even numeric initial data in the input file like this:

```yaml
InitialData:
  Sinusoid:  # or Krivodonova, Kuzmin, Numeric
```

This is possible with our factory-creation infrastructure, that you have already encountered in the `factory_creation` struct in the metavariables. You would inherit the initial-data classes from a base class like `InitialData` and add an entry `std::pair<InitialData, tmpl::list</* all possible choices */>>` to the factory-creation struct. We'll have to iron out a few details, but since you have written the entire scalar advection system yourself I think you have plenty of experience to manage this :) If @kidder agrees this is a good idea we can set up a call with whoever else is interested to discuss the design.

Reply:
Yes this is fine, and to address @yoonso0-0 's concern, this is not something that needs to be done very soon so you can take your time and become more familiar with the code.

Reply:
Okay!